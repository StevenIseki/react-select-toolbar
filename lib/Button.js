'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _templateObject = _taggedTemplateLiteral(['\n  display: inline-block;\n  margin: 0 !important;\n  color: ', ';\n  &:hover {\n    color: ', ';\n    span {\n      opacity: 1;\n    }\n  }\n  &:active {\n    color: #ccc;\n  }\n'], ['\n  display: inline-block;\n  margin: 0 !important;\n  color: ', ';\n  &:hover {\n    color: ', ';\n    span {\n      opacity: 1;\n    }\n  }\n  &:active {\n    color: #ccc;\n  }\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  padding: 0;\n  cursor: pointer;\n  height: 40px;\n  width: 40px;\n  background: transparent;\n  margin-top: 0.4rem;\n  margin-bottom: -1rem;\n'], ['\n  padding: 0;\n  cursor: pointer;\n  height: 40px;\n  width: 40px;\n  background: transparent;\n  margin-top: 0.4rem;\n  margin-bottom: -1rem;\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n  display: block;\n  position: fixed;\n  overflow: hidden;\n  font-size: 12px;\n  margin-top: -3.3rem;\n  text-transform: uppercase;\n  opacity: 0;\n  color: black !important;\n  transition: opacity 0.5s ease-in-out;\n  z-index: 100;\n  padding: 0.2rem 0.5rem;\n  border: 1px solid #F1F1F1;\n  border-radius: 2px;\n  background: white;\n  box-shadow: 3px 3px 5px #BFBDBD;\n'], ['\n  display: block;\n  position: fixed;\n  overflow: hidden;\n  font-size: 12px;\n  margin-top: -3.3rem;\n  text-transform: uppercase;\n  opacity: 0;\n  color: black !important;\n  transition: opacity 0.5s ease-in-out;\n  z-index: 100;\n  padding: 0.2rem 0.5rem;\n  border: 1px solid #F1F1F1;\n  border-radius: 2px;\n  background: white;\n  box-shadow: 3px 3px 5px #BFBDBD;\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _class = function (_Component) {
  _inherits(_class, _Component);

  function _class() {
    _classCallCheck(this, _class);

    return _possibleConstructorReturn(this, (_class.__proto__ || Object.getPrototypeOf(_class)).apply(this, arguments));
  }

  _createClass(_class, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var Button = this.props.item.icon;
      var action = this.props.item.action;


      return _react2.default.createElement(
        ToolbarButtonWrapper,
        null,
        _react2.default.createElement(
          Label,
          null,
          this.props.item.label
        ),
        _react2.default.createElement(
          ToolbarButton,
          { onClick: function onClick() {
              return action(_this2.props);
            } },
          _react2.default.createElement(Button, null)
        )
      );
    }
  }]);

  return _class;
}(_react.Component);

exports.default = _class;


var ToolbarButtonWrapper = _styledComponents2.default.li(_templateObject, function (props) {
  return props.active ? '#f00' : '#000';
}, function (props) {
  return props.active ? '#fcc' : '#ccc';
});

var ToolbarButton = _styledComponents2.default.div(_templateObject2);

var Label = _styledComponents2.default.span(_templateObject3);