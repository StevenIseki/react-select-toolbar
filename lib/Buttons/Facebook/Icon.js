'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _templateObject = _taggedTemplateLiteral(['\n  transform: scale(0.8);\n'], ['\n  transform: scale(0.8);\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _class = function (_React$Component) {
  _inherits(_class, _React$Component);

  function _class() {
    _classCallCheck(this, _class);

    return _possibleConstructorReturn(this, (_class.__proto__ || Object.getPrototypeOf(_class)).apply(this, arguments));
  }

  _createClass(_class, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        Icon,
        { height: '24', width: '24', fill: 'currentColor', viewBox: '0 0 430.113 430.114' },
        _react2.default.createElement('path', { d: 'M158.08 83.3v59.218h-43.384v72.412h43.385v215.183h89.123V214.936h59.805s5.6-34.72 8.316-72.685H247.54V92.74c0-7.4 9.717-17.354 19.32-17.354h48.558V0h-66.02c-93.52-.004-91.317 72.48-91.317 83.3z' })
      );
    }
  }]);

  return _class;
}(_react2.default.Component);

exports.default = _class;


var Icon = _styledComponents2.default.svg(_templateObject);