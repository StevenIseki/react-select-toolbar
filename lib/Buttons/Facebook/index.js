"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var action = function action(props) {
  console.log(props);
};

var _default = action;
exports.default = _default;