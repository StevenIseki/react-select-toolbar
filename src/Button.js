import React, {Component} from 'react'
import styled from 'styled-components'

export default class extends Component {
  render () {
    const Button = this.props.item.icon
    const {action} = this.props.item

    return (
      <ToolbarButtonWrapper>
        <Label>{this.props.item.label}</Label>
        <ToolbarButton onClick={() => action(this.props)}>
          <Button />
        </ToolbarButton>
      </ToolbarButtonWrapper>
    )
  }
}

const ToolbarButtonWrapper = styled.li`
  display: inline-block;
  margin: 0 !important;
  color: ${props => props.active ? '#f00' : '#000'};
  &:hover {
    color: ${props => props.active ? '#fcc' : '#ccc'};
    span {
      opacity: 1;
    }
  }
  &:active {
    color: #ccc;
  }
`

const ToolbarButton = styled.div`
  padding: 0;
  cursor: pointer;
  height: 40px;
  width: 40px;
  background: transparent;
  margin-top: 0.4rem;
  margin-bottom: -1rem;
`

const Label = styled.span`
  display: block;
  position: fixed;
  overflow: hidden;
  font-size: 12px;
  margin-top: -3.3rem;
  text-transform: uppercase;
  opacity: 0;
  color: black !important;
  transition: opacity 0.5s ease-in-out;
  z-index: 100;
  padding: 0.2rem 0.5rem;
  border: 1px solid #F1F1F1;
  border-radius: 2px;
  background: white;
  box-shadow: 3px 3px 5px #BFBDBD;
`
