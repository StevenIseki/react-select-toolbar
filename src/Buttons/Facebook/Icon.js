import React from 'react'
import styled from 'styled-components'

export default class extends React.Component {
  render () {
    return (
      <Icon height='24' width='24' fill='currentColor' viewBox='0 0 430.113 430.114'>
        <path d='M158.08 83.3v59.218h-43.384v72.412h43.385v215.183h89.123V214.936h59.805s5.6-34.72 8.316-72.685H247.54V92.74c0-7.4 9.717-17.354 19.32-17.354h48.558V0h-66.02c-93.52-.004-91.317 72.48-91.317 83.3z' />
      </Icon>
    )
  }
}

const Icon = styled.svg`
  transform: scale(0.8);
`
