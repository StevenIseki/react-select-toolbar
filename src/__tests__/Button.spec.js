/* setup enzyme */
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

/* setup jsdom */
var jsdom = require('jsdom')
const { JSDOM } = jsdom
const window = new JSDOM('').window
global.window = window
global.document = window.document

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Button from '../Button'
import FacebookIcon from '../Buttons/Facebook/Icon'
import Facebook from '../Buttons/Facebook'

test('Button renders correctly and matches snapshot', () => {
  const handleChange = jest.fn()
  const handleClick = jest.fn()
  const component = renderer.create(
    <Button key={0} item={{label: 'facebook', icon: FacebookIcon, action: Facebook}} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Button renders the correct elements and props', () => {
  const ITEM = {label: 'facebook', icon: FacebookIcon, action: Facebook};
  const wrapper = shallow(
    <Button key={0} item={ITEM} />
  )

  expect(wrapper.instance().props.item).toEqual(ITEM);

  expect(
    wrapper
      .text()
  ).toContain('facebook')

  {/* console.log(wrapper.debug()) */}
})
