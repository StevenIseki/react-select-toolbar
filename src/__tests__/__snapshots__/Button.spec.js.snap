// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly and matches snapshot 1`] = `
<li
  className="sc-bdVaJa cFyHDk"
>
  <span
    className="sc-htpNat fHQAuj"
  >
    facebook
  </span>
  <div
    className="sc-bwzfXH inRAPE"
    onClick={[Function]}
  >
    <svg
      className="sc-bxivhb cZkvWD"
      fill="currentColor"
      height="24"
      viewBox="0 0 430.113 430.114"
      width="24"
    >
      <path
        d="M158.08 83.3v59.218h-43.384v72.412h43.385v215.183h89.123V214.936h59.805s5.6-34.72 8.316-72.685H247.54V92.74c0-7.4 9.717-17.354 19.32-17.354h48.558V0h-66.02c-93.52-.004-91.317 72.48-91.317 83.3z"
      />
    </svg>
  </div>
</li>
`;
